### PR TITLE
[KYV-759] Created required DTO:s and mappers to enable serialization …

### PR DESCRIPTION
**Pull request by martinldev**
Created required DTO:s and mappers to enable serialization of payment creation response and added a simple test for shop-in-shop payment creation.

### CODE REVIEW

File: core/src/main/java/org/helsinki/paytrail/mapper/PaytrailPaymentCreateResponseMapper.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        try {
+            to.setPaymentResponse(mapper.readValue(paytrailResponse.getResultJson(), new TypeReference<>() {
+            }));
+        } catch (JsonProcessingException e) {
+            String[] errors = new String[]{
+                    e.getMessage(),
+                    e.getOriginalMessage(),
+                    paytrailResponse.getResultJson()
+            };
+            to.setErrors(errors);
```
Comment (severikupari1) [inline core/src/main/java/org/helsinki/paytrail/mapper/PaytrailPaymentCreateResponseMapper.java:28]: Should you log this error too?

File: core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentResponse.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@Data
+@Accessors(chain = true)
+public class PaytrailPaymentResponse implements Serializable {
+    private String transactionId;
+    private String href;
+    private String terms;
+    private List<PaymentMethodGroupData> groups;
+    private List<Provider> providers;
+    private String reference;
+    private JsonNode customProviders;
```
Comment (severikupari1) [inline core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentResponse.java:20]: Could this be some kind of dto?

File: core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java [hunk 3]
```diff
@@ -5,73 +5,102 @@
...
 
-        payload.setStamp("d2568f2a-e4c6-40ba-a7cd-d573382ce548");
+        try {
+            Gson gson = new Gson();
+            String json = gson.toJson(response.get());
+            System.out.println(json);
+            System.out.println(paymentCreateResponse.toString());
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
+        }
```
Comment (severikupari1) [inline core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java:74]: Could you add some assetions to returning json? Like the values you give are returned in response?

File: core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java [hunk 4]
```diff
@@ -82,20 +111,19 @@ void createTestPayment() throws ExecutionException, InterruptedException {
...
-//            String json = gson.toJson(response.get());
-//            System.out.println(json);
-//        } catch (InterruptedException | ExecutionException e) {
-//            throw new RuntimeException(e);
-//        }
+        PaytrailPaymentCreateRequest request = new PaytrailPaymentCreateRequest(payload);
+        CompletableFuture<PaytrailPaymentCreateResponse> response = client.sendRequest(request);
+
+        PaytrailPaymentCreateResponse paymentCreateResponse = mapper.to(response.get());
+
```
Comment (severikupari1) [inline core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java:118]: Could you add some assertions to returning json? Like the values you give are returned in response?

File: core/src/main/java/org/helsinki/paytrail/mapper/PaytrailPaymentCreateResponseMapper.java (hunk 1 above)
Comment (martinldev) [reply core/src/main/java/org/helsinki/paytrail/mapper/PaytrailPaymentCreateResponseMapper.java:28]: Well the errors are added to the returned response object, but sure I could add some logging also.

File: core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java (hunk 4 above)
Comment (martinldev) [reply core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java:118]: Sure!

File: core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentResponse.java (hunk 2 above)
Comment (martinldev) [reply core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentResponse.java:20]: It is a DTO like all these other classes under model/payments. Maybe I could just rename the class to PaytrailPaymentTransaction.

File: core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentTransaction.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.experimental.Accessors;
+
+import java.io.Serializable;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+@Data
+@Accessors(chain = true)
+public class PaytrailPaymentTransaction implements Serializable {
```
Comment (severikupari1) [inline core/src/main/java/org/helsinki/paytrail/model/payments/PaytrailPaymentTransaction.java:13]: Change back to PaytrailPaymentReponse

File: core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java (hunk 3 above)
Comment (martinldev) [reply core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java:81]: Added some basic assertions, but returning data does not contain any of the provided request params, so there is no way to reflect returning data on request data.

**Pull request merged**